GramAddict/plugins/interact_hashtag_posts.py

```diff
@@ -0,0 +1,264 @@
+    ):
+        interaction = partial(
+            interact_with_user,
+            my_username=self.session_state.my_username,
+            likes_count=likes_count,
+            stories_count=stories_count,
+            stories_percentage=stories_percentage,
+            follow_percentage=follow_percentage,
+            on_like=on_like,
+            on_watch=on_watch,
+            profile_filter=profile_filter,
+            args=self.args,
+            session_state=self.session_state,
+            current_mode=self.current_mode,
+        )
+
+        is_follow_limit_reached = partial(
+            is_follow_limit_reached_for_source,
+            follow_limit=follow_limit,
+            source=hashtag,
+            session_state=self.session_state,
+        )
+        search_view = TabBarView(device).navigateToSearch()
+        if not search_view.navigateToHashtag(hashtag):
+            return
+
+        logger.info("Switching to Recent tab")
+        HashTagView(device)._getRecentTab().click()
+        random_sleep(5, 10)
+        if HashTagView(device)._check_if_no_posts():
+            HashTagView(device)._reload_page()
+            random_sleep(4, 8)
+
+        logger.info("Opening the first result")
+
+        result_view = HashTagView(device)._getRecyclerView()
+        HashTagView(device)._getFistImageView(result_view).click()
+        random_sleep()
+
+        posts_end_detector = ScrollEndDetector(repeats_to_end=2)
+        post_description = ""
+
+        def interact():
+            can_follow = not is_follow_limit_reached() and (
+                storage.get_following_status(username) == FollowingStatus.NONE
+                or storage.get_following_status(username) == FollowingStatus.NOT_IN_LIST
+            )
+
+            interaction_succeed, followed = interaction(
+                device, username=username, can_follow=can_follow
+            )
+            storage.add_interacted_user(username, followed=followed)
+            can_continue = on_interaction(
+                succeed=interaction_succeed, followed=followed
+            )
+            if not can_continue:
+                return
+
+        def random_choice():
+            from random import randint
+
+            random_number = randint(1, 100)
+            if interact_chance > random_number:
+                return True
+            else:
+                return False
+
+        while True:
+            if random_choice():
+                username = PostsViewList(device)._get_post_owner_name()[:-3]
+                if storage.is_user_in_blacklist(username):
+                    logger.info(f"@{username} is in blacklist. Skip.")
+                elif storage.check_user_was_interacted(username):
+                    logger.info(f"@{username}: already interacted. Skip.")
+                else:
+                    logger.info(f"@{username}: interact")
+                    PostsViewList(device)._like_in_post_view()
+                    # if random_choice():
```

Comment:
?

Reply:
i leave an hint for doing things directly from feed. 
like comments and follow

Reply:
Please document the purpose and how to use it.